### PR TITLE
more details in examples

### LINKED ISSUE #95: Add explanatory test to make the documentation examples more understandable
**Feature request**: Add more helpful motivating and descriptive language to the 3 documentation examples

The examples in the documentation could use some more descriptive language to describe what is being demonstrated. I put in a [PR](https://github.com/rcaneill/xnemogcm/pull/93) to help get started with this, but the examples have very few descriptions of what is going on and, importantly, why it's a useful example. A couple sentences at the top of each of the 3 examples (along the lines I suggest in my PR when describing the examples) would be very helpful for users to successfully follow along with those examples.

For example, in the first example `open_process_files`:
- The example starts by saying "Start by importing the functions." I suggest a sentence or two before this should be added to let the user know exactly what this example will demonstrate.
- The term "process" here is not clear to me. Processing data can take many forms. If possible, either at the top of the example or later when discussing the steps to "process" files, please explain what type of processing occurs and why this is useful for the user.
- The first section title in this example is "First open the domain_cfg and nemo files into 2 datasets". Could you better explain what a `domain_cfg` file is and why we need to open that and the "Nemo files" as 2 datasets? 

These are just a few examples - in general, a bit more explanatory text would be helpful at each main step of each example.

Apologies that I can't easily put in a PR for this, since I don't know enough about NEMO to do this well. Hopefully this will be relatively straight-forward for a NEMO and `xnemogcm` expert like you @rcaneill! 😊

### PR DESCRIPTION
**Pull request by rcaneill**
closes #95

### CODE REVIEW

File: docs/examples/compute_metrics.ipynb [hunk 1]
```diff
@@ -1,5 +1,14 @@
 {
  "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "f95face2-9fbc-4ec5-8546-ce17fe290147",
+   "metadata": {},
+   "source": [
+    "This example showcases how `xnemogcm` can compute certain missing metrics (metrics are called scale factors in the NEMO community, and called metrics in the xgcm community).\n",
```
Comment (paigem) [inline docs/examples/compute_metrics.ipynb:8]: Thanks for adding this sentence, about how metrics are called scale factors in NEMO, and metrics in xgcm - this is very helpful!

File: docs/examples/compute_metrics.ipynb [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+    "It is thus possible to e.g. compute `e3u` from `e3t`. We let the reader refer to the NEMO documentation for the explanation of the scale factors."
```
Comment (paigem) [inline docs/examples/compute_metrics.ipynb:9]: Can you add a link directly to the NEMO documentation about scale factors here?

**Review by paigem — COMMENTED** [review]
One minor suggestion to include a link directly to the NEMO documentation if easy to do. Otherwise, this looks great - thanks!

File: docs/examples/compute_metrics.ipynb (hunk 2 above)
Comment (rcaneill) [reply docs/examples/compute_metrics.ipynb:9]:
```suggestion
    "It is thus possible to e.g. compute `e3u` from `e3t`. We let the reader refer to the NEMO documentation for the explanation of the scale factors, https://doi.org/10.5281/zenodo.8167700."
```

Comment (rcaneill) [reply docs/examples/compute_metrics.ipynb:9]: I'll do it in another PR

**Pull request merged**